std/stdio.d

```diff
@@ -4522,70 +4521,70 @@ Initialize with a message and an error code.
     }
 }
 
-extern(C) void std_stdio_static_this()
+// Undocummented but public because std* handles are aliasing it
+ref File makeGlobal(alias handle)()
 {
-    static import core.stdc.stdio;
-    //Bind stdin, stdout, stderr
-    __gshared File.Impl stdinImpl;
-    stdinImpl.handle = core.stdc.stdio.stdin;
-    .stdin._p = &stdinImpl;
-    // stdout
-    __gshared File.Impl stdoutImpl;
-    stdoutImpl.handle = core.stdc.stdio.stdout;
-    .stdout._p = &stdoutImpl;
-    // stderr
-    __gshared File.Impl stderrImpl;
-    stderrImpl.handle = core.stdc.stdio.stderr;
-    .stderr._p = &stderrImpl;
+    static __gshared File.Impl impl;
+    static __gshared File result;
+    static shared bool initialized;
+    if (!initialized)
+    {
+        // Use double-checking because initOnce is inefficient.
```

Comment:
It's not third, it's second, the same it was before.

I tripped over that at first too. initOnce() internally uses the double-checking pattern with atomicLoad, which may or may not be more expensive than a plain load depending on platform. But 'true' can never be observed on this new extra check before initOnce() has completed, so in effect this simply short-circuits a potentially more expensive load. It may come at a cost of a few extra calls to initOnce() though, depending on how memory works on a particular platform.
If it was allowed for `initialized` to flip from `true` to `false` during runtime, that new `if` would indeed have been a mistake.

IIRC, TDPL initially proposed that reads and writes from/to `shared` should be both atomic **and** fenced. That last part is a very heavy restriction though, and, among other things, shows exactly why we need a DIP to make `shared` actually work.